code/controllers/subsystem/nightshift.dm

```diff
@@ -0,0 +1,66 @@
+SUBSYSTEM_DEF(nightshift)
+	name = "Night shift"
+	wait = 600
+	flags = SS_BACKGROUND
+
+	var/nightshift_active = FALSE
+	var/nightshift_start_time = 702000		//7:30 PM, station time
+	var/nightshift_end_time = 270000		//7:30 AM, station time
+	var/nightshift_first_check = 3000		//5 minutes, 60 * 5 * 10 ds, world time
+
+	var/obey_security_level = TRUE
+	var/high_security_mode = FALSE
+
+/datum/controller/subsystem/nightshift/Initialize()
+	if(!CONFIG_GET(flag/enable_night_shifts))
+		can_fire = FALSE
+	. = ..()
+
+/datum/controller/subsystem/nightshift/fire(resumed = FALSE)
+	check_nightshift()
+
+/datum/controller/subsystem/nightshift/proc/check_nightshift(force_set = FALSE)
+	var/time = station_time()
+	var/nightshift = time < nightshift_end_time && time > nightshift_start_time

```

Comment:
Try implementing the 1 day cycle = 45 minutes idea I had

Reply:
done